packages/l/libcurl/xmake.lua

```diff
@@ -73,7 +73,7 @@ package("libcurl")
         local has_deps = false
         for name, dep in pairs(configdeps) do
             if package:config(name) then
-                package:add("deps", dep, {host = package:is_binary()})
+                package:add("deps", dep, {host = package:is_binary(), configs = {shared = package:config("shared")}})
```

Comment:
why? what's issue is fixed?

Reply:
With the following xmake.lua
```
add_requireconfs("*", {configs = {shared = true}})
add_requires("libcurl", {system = false, configs = {shared = true, mbedtls = true , openssl = false }})

target("test-curl")
       set_kind("binary")
       add_packages("libcurl")
```
When build with gcc-arm , a lot of errors

`xmake f -vyc -p cross --sdk=/opt/cross/arm-none-linux-gnueabihf/`
```
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o):(.rodata+0x26c0): dangerous relocation: unsupported relocation
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o):(.rodata+0x26d0): dangerous relocation: unsupported relocation
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o):(.rodata+0x26e0): dangerous relocation: unsupported relocation
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o):(.rodata+0x26f0): dangerous relocation: unsupported relocation
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o):(.rodata+0x2700): dangerous relocation: unsupported relocation
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o):(.rodata+0x2710): dangerous relocation: unsupported relocation
/opt/cross/arm-none-linux-gnueabihf/bin/../lib/gcc/arm-none-linux-gnueabihf/11.3.1/../../../../arm-none-linux-gnueabihf/bin/ld: /home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_tls.c.o)(.text+0x5020): unresolvable R_ARM_THM_CALL relocation against symbol `printf@@GLIBC_2.4'
collect2: error: ld returned 1 exit status
make[2]: *** [lib/CMakeFiles/libcurl_shared.dir/build.make:2790: lib/libcurl.so.4.8.0] Error 1
```

Reply:
a lot of errors 
```
/opt/cross/arm-none-linux-gnueabihf/bin/../lib/gcc/arm-none-linux-gnueabihf/11.3.1/../../../../arm-none-linux-gnueabihf/bin/ld: /home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedcrypto.a(ecjpake.c.o): relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/opt/cross/arm-none-linux-gnueabihf/bin/../lib/gcc/arm-none-linux-gnueabihf/11.3.1/../../../../arm-none-linux-gnueabihf/bin/ld: /home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedcrypto.a(hmac_drbg.c.o): relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/opt/cross/arm-none-linux-gnueabihf/bin/../lib/gcc/arm-none-linux-gnueabihf/11.3.1/../../../../arm-none-linux-gnueabihf/bin/ld: /home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedcrypto.a(pkcs12.c.o): relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/opt/cross/arm-none-linux-gnueabihf/bin/../lib/gcc/arm-none-linux-gnueabihf/11.3.1/../../../../arm-none-linux-gnueabihf/bin/ld: /home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedcrypto.a(pkcs5.c.o): relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/opt/cross/arm-none-linux-gnueabihf/bin/../lib/gcc/arm-none-linux-gnueabihf/11.3.1/../../../../arm-none-linux-gnueabihf/bin/ld: /home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedcrypto.a(poly1305.c.o): relocation R_ARM_THM_MOVW_ABS_NC against `a local symbol' can not be used when making a shared object; recompile with -fPIC
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o): in function `mbedtls_ssl_list_ciphersuites':
ssl_ciphersuites.c:(.text+0xcc): dangerous relocation: unsupported relocation
/home/wps/.xmake/packages/m/mbedtls/v3.6.0/54603c366b9b472ea75e0246e04819e3/lib/libmbedtls.a(ssl_ciphersuites.c.o): in function `mbedtls_ssl_get_ciphersuite_sig_pk_alg':
ssl_ciphersuites.c:(.text+0x1e8): dangerous relocation: unsupported relocation
ssl_ciphersuites.c:(.text+0x1ec): dangerous relocation: unsupported relocation
```

Reply:
it's fPIC issue of mbedtls. you should not change all deps. 

you can debug and improve mbedtls to enable -fPIC for static library.

Reply:
it's not the issue of -fPIC in mbedtls.  Just modify libcurl, everything works fine.
I've tested on Android NDK R26D, linux gcc-x64, linux gcc-arm32. works fine.

Reply:
I renamed libcurl to yw_libcurl 

before my modify, the command line to cmake is

```
-DMBEDTLS_LIBRARY=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/695b3854d4e940f5ab527eaa77168f2c/lib/libmbedtls.a 
-DMBEDX509_LIBRARY=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/695b3854d4e940f5ab527eaa77168f2c/lib/libmbedx509.a 
-DMBEDCRYPTO_LIBRARY=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/695b3854d4e940f5ab527eaa77168f2c/lib/libmbedcrypto.a 
```

After my modify
```
        -DMBEDTLS_INCLUDE_DIRS=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/390e1402569e47a8a034b7fa1d79eeab/include 
        -DMBEDTLS_LIBRARY=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/390e1402569e47a8a034b7fa1d79eeab/lib/libmbedtls.so 
        -DMBEDX509_LIBRARY=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/390e1402569e47a8a034b7fa1d79eeab/lib/libmbedx509.so 
        -DMBEDCRYPTO_LIBRARY=/home/wps/.xmake/packages/y/yw_mbedtls/20240703.105529/390e1402569e47a8a034b7fa1d79eeab/lib/libmbedcrypto.so 
        -DCMAKE_INSTALL_PREFIX=/home/wps/.xmake/packages/y/yw_libcurl/8.8.0/53fe5c3a804a4835a86113af15fd7a5f -DCMAKE_INSTALL_LIBDIR:PATH=lib 
```

Reply:
Using dynamic library dependencies certainly solves the problem, but xmake-repo's packages use static library dependencies by default. If you change the dependency configuration of libcurl, it will easily lead to dependency conflicts.

And it doesn't solve the problem of libcurl using the static library mbedtls. If we want to fix this issue, we should debug mbedtls package.

You should configure your project to use dynamic dependencies instead of modifying the libcurl package.

```lua
add_requires("libcurl")
add_requireconfs("libcurl.**", {override = true, configs = {shared = true}})
```

Reply:
If you don't approve this PR, just ignore it.
I'll using my private version of xmake.lua to resolve my problem.

Reply:
why not use `add_requireconfs("libcurl.**", {override = true, configs = {shared = true}})`

only one line, it will resolve your problem.